(EAI-200): Add tool calling to MongoDB Chat Server (#318)
mongodb/chatbot: docs/docs/server/tools.md

and whether the user query should be rejected.

```ts
// weatherTool.ts

import { Tool } from "mongodb-chatbot-server";
export const weatherTool: Tool = {
  definition: {
    name: "getCurrentWeather",
    description: "Get the weather in location",
    parameters: {
      type: "object",
      properties: {
        location: {
          type: "string",
          description: "The city and state e.g. San Francisco, CA",
        },
        unit: { type: "string", enum: ["c", "f"] },
      },
      required: ["location"],
    },
  },
  async call({ functionArgs: { location, unit } }) {
    const weatherReport = await fetchWeatherFromApi(location, unit);
    return {
      toolCallMessage: {
        role: "function",
        name: "getCurrentWeather",
        content: `The weather in ${location} is: ${weatherReport.content}`,
      },
      references: [{ title: `${location} weather`, url: weatherReport.url }],
    };
  },
};
```

## Give Tools to the ChatLLM

To give the chatbot access to the tool, you must provide the [`ChatLlm`](../reference/server/interfaces/ChatLlm.md) with the tool.

The [`makeOpenAiChatLlm()`](../reference/server/modules.md#makeopenaichatllm)

> function, which you can use to chat with an OpenAI model, takes a `tools` argument.

Reviewer: ```suggestion
function, which you can use to instantiate an OpenAI model, takes a `tools` argument.
```

Author: ```suggestion
function, which you can use to instantiate a client for an OpenAI model, takes a `tools` argument.
```
